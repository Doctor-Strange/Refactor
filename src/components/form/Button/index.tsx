import React from "react";
import "./Button.scss";

const Button = (props: IButton) => {
  return (
    <button
    data-test-id="btn"
      className={props.class}
      disabled={props.disable || props.loading}
      onClick={() => props.click()}
    >
      {props.loading ? <span  data-test-id="Loading" className="Loading" /> : props.value}
    </button>
  );
};

interface IButton {
  value: string;
  class: string;
  loading: boolean;
  disable?: boolean;
  click: any;
}

export default Button;
