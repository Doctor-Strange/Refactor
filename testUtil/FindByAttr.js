

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
