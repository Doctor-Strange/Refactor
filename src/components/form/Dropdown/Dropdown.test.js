import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../../testUtil/FindByAttr";
import Button from "./";

const setup = (props) => {
  return mount(<Button {...props} />);
};

const clearFieldMock = jest.fn();
const SelectMock = jest.fn();

const props = {
  data: [
    { text: "a", value: "a" },
    { text: "b", value: "b" },
  ],
  clearField: clearFieldMock,
  Select: SelectMock,
};

describe("/Dropdown component", () => {
  let wrapper;
  describe("To match snapshot", () => {
    test("dropDown Snapshot", () => {
      wrapper = setup(props);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
