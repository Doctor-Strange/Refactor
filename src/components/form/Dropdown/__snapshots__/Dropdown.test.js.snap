// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Dropdown component To match snapshot dropDown Snapshot 1`] = `ReactWrapper {}`;
